carapace-server/src/main/java/org/carapaceproxy/server/cache/ContentsCache.java

```diff
@@ -154,14 +154,22 @@ private boolean isCachable(HttpRequest request, boolean registerNoCacheStat) {
             default:
                 return false;
         }
-        boolean ctrlF5 = request.headers()
-                .containsValue(HttpHeaderNames.CACHE_CONTROL, HttpHeaderValues.NO_CACHE, true);
+
+        final HttpHeaders headers = request.headers();
+        final String cacheControl = headers.get(HttpHeaderNames.CACHE_CONTROL, "").replaceAll(" ", "").toLowerCase();
+        boolean ctrlF5 = cacheControl.contains(HttpHeaderValues.NO_CACHE);
         if (ctrlF5) {
             if (registerNoCacheStat) {
                 NO_CACHE_REQUESTS_COUNTER.inc();
             }
             return false;
         }
+        if ((secure && !cacheControl.contains(HttpHeaderValues.PUBLIC)
+                || headers.contains(HttpHeaderNames.PRAGMA, HttpHeaderValues.NO_CACHE, true))) {
```

Comment:
please handle case sensitive, and when we have header with multiple part like Cache-Control: public, max-age 3600

Reply:
This is how actually works

Reply:
ok look good